packages/dds/tree/src/feature-libraries/modular-schema/comparison.ts

```diff
@@ -203,6 +213,192 @@ export function allowsRepoSuperset(
 	return true;
 }
 
+/**
+ * @returns the discrepancies between two TreeStoredSchema objects
+ */
+export function allowsDocumentConcepts(
+	original: TreeStoredSchema,
+	updated: TreeStoredSchema,
+): Incompatibility[] {
+	const incompatibilities: Incompatibility[] = [];
+	const areSetsEqual = (a: TreeTypeSet, b: TreeTypeSet): boolean => {
+		if (a === undefined && b === undefined) {
+			return true;
+		}
+		if (a !== undefined && b !== undefined) {
+			return a.size === b.size && [...a].every((value) => b.has(value));
+		}
+		return false;
+	};
+	const getFieldKind = (x: unknown): SchemaFactoryFieldKind | undefined => {
```

Comment:
this conversion is silly, we should just use consistent terminology to how the schema is defined. Update required/optional/array to use value/optional/sequence where required, IMO.